packages/flatpak.rb

```diff
@@ -0,0 +1,90 @@
+require 'package'
+
+class Flatpak < Package
+  description 'Flatpak is a system for building, distributing, and running sandboxed desktop applications on Linux.'
+  homepage 'https://flatpak.org'
+  @_ver = '1.10.1'
+  version @_ver
+  compatibility 'all'
+  source_url "https://github.com/flatpak/flatpak/releases/download/#{@_ver}/flatpak-#{@_ver}.tar.xz"
+  source_sha256 'c1354f42bf3b5d51aeb4028c9b62fd4ffc673ef2ff6e583c17777f5dafdbdcb7'
+
+  binary_url({
+    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/flatpak-1.10.1-chromeos-armv7l.tar.xz',
+     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/flatpak-1.10.1-chromeos-armv7l.tar.xz',
+       i686: 'https://dl.bintray.com/chromebrew/chromebrew/flatpak-1.10.1-chromeos-i686.tar.xz',
+     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/flatpak-1.10.1-chromeos-x86_64.tar.xz'
+  })
+  binary_sha256({
+    aarch64: 'ee4cf9fe437d1aa7b6804d0b323c81c9df25cd14a6124bc5d624ad3238df6875',
+     armv7l: 'ee4cf9fe437d1aa7b6804d0b323c81c9df25cd14a6124bc5d624ad3238df6875',
+       i686: '6cfa3ece1b485369f16fb62c5b17744d56a7aa162a95aab49d3d9663026cbd60',
+     x86_64: 'ac8e6a92a18862201c12563036f206804547d2a1d350f548d130bb38b9dfea5b'
+  })
+
+  depends_on 'bubblewrap'
+  depends_on 'libostree'
+  depends_on 'polkit'
+  depends_on 'appstream_glib'
+  depends_on 'libseccomp'
+  depends_on 'pyparsing'
+  depends_on 'dconf'
+
+  def self.patch
+    patch_description = 'backported fixes'
+    patch_url = 'https://patch-diff.githubusercontent.com/raw/flatpak/flatpak/pull/4132.patch'
+    patch_sha256 = '8230491be909a027fdb5a03ff0fa638331c940289fa6d68a21489d78c6f7f6b0'
+    patch_filename = 'patch'
```

Comment:
My general rule is you don't need a variable assigned unless you plan on using it more than once.  I could probably make the argument you don't need the patch_filename variable either.

Reply:
So I'm trying to use a common code block for patching between packages.

Ideally there would be a way to add multiple patches without too much code reuse.

Case in point: We should use MOST of these patches for libsdl: https://github.com/archlinux/svntogit-packages/tree/packages/sdl/trunk

@supechicken Any chance you want to play with adding a patch framework to crew so we don't have to reinvent this for every single package which needs patches?

Reply:
Examples of how patches are put into package files elsewhere it would be nice to repicate:

Arch: https://github.com/archlinux/svntogit-packages/blob/packages/sdl/trunk/PKGBUILD
Homebrew: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/sdl.rb

(I think the arch version is more elegant, but the homebrew method might be easier to copy since their code is in Ruby.)

Reply:
> So I'm trying to use a common code block for patching between packages.
> 
> Ideally there would be a way to add multiple patches without too much code reuse.
> 
> Case in point: We should use MOST of these patches for libsdl: https://github.com/archlinux/svntogit-packages/tree/packages/sdl/trunk
> 
> @supechicken Any chance you want to play with adding a patch framework to crew so we don't have to reinvent this for every single package which needs patches?

I will take a try at it :)